docs/rules/no-get-by-for-absent-elements.md

```diff
@@ -1,75 +1,62 @@
 - using `getBy` queries as an assert itself, so if the element is not found the error thrown will work as the check itself within the test.
 
-However, when trying to assert if an element is not present or disappearance, we can't use `getBy*` as the test will fail immediately. Instead it is recommended to use `queryBy*`, which does not throw and therefore we can assert that e.g. `expect(queryByText("Foo")).not.toBeInTheDocument()`.
+However, when trying to assert if an element is not present or disappearance, using `getBy*` will make the test fail immediately. Instead it is recommended to use `queryBy*`, which does not throw and therefore we can assert that e.g. `expect(queryByText("Foo")).not.toBeInTheDocument()`.
 
 > The same applies for the `getAll*` and `queryAll*` queries too.
 
 ## Rule details
 
-This rule gives a notification whenever `expect` is used with one of the query functions that throw an error if the element is not found.
+This rule gives a notification whenever `expect` is used with one of the query functions that throws an error if the element is not found.
 
 Examples of **incorrect** code for this rule:
 
 ```js
 test('some test', () => {
   const { getByText, getAllByText } = render(<App />);
-  expect(getByText('Foo')).toBeInTheDocument();
-  expect(getAllByText('Foo')[0]).toBeInTheDocument();
   expect(getByText('Foo')).not.toBeInTheDocument();
-  expect(getAllByText('Foo')[0]).not.toBeInTheDocument();
+  expect(getAllByText('Foo')[0]).toBeNull();
+  expect(getByText('Foo')).toBeFalsy();
+  expect(getAllByText('Foo')[0]).not.toBeTruthy();
+  expect(getByText('Foo')).toBeUndefined();
 });
 ```
 
 ```js
 test('some test', async () => {
   const utils = render(<App />);
-  await wait(() => {
+  await waitForElementToBeRemoved(() => {
     expect(utils.getByText('Foo')).toBeInTheDocument();
   });
-  await wait(() => {
+  await waitForElementToBeRemoved(() => {
     expect(utils.getAllByText('Foo')).toBeInTheDocument();
   });
-  await wait(() => {
-    expect(utils.getByText('Foo')).not.toBeInTheDocument();
-  });
-  await wait(() => {
-    expect(utils.getAllByText('Foo')).not.toBeInTheDocument();
-  });
 });
 ```
 
 Examples of **correct** code for this rule:
 
 ```js
 test('some test', () => {
-  const { queryByText, queryAllByText } = render(<App />);
-  expect(queryByText('Foo')).toBeInTheDocument();
-  expect(queryAllByText('Foo')[0]).toBeInTheDocument();
-  expect(queryByText('Foo')).not.toBeInTheDocument();
-  expect(queryAllByText('Foo')[0]).not.toBeInTheDocument();
+  const { getByText, getAllByText } = render(<App />);
+  expect(getByText('Foo')).toBeInTheDocument();
+  expect(getAllByText('Foo')).toBeTruthy();
 });
 ```
 
 ```js
 test('some test', async () => {
   const utils = render(<App />);
-  await wait(() => {
+  await waitForElementToBeRemoved(() => {
     expect(utils.queryByText('Foo')).toBeInTheDocument();
```

Comment:
Shouldn't this be (same below)

```suggestion
    return utils.queryByText('Foo')
```

Reply:
Indeed. There are two cases for this rule as mentioned in the issue: asserting elements are not present (rule must check this when not.toBeInTheDocument() or toBeNull() found) or waiting for element to be removed (this is not an assertion but an async util that must no use getBy inside).

This is also why I suggested “absence” in the rule so it does not refer directly to any of the motivation for the rule itself (“not present” and “disappearing”)

Reply:
I can elaborate more tomorrow as I’m on mobile now